pkg/operator/baremetal_pod.go

```diff
@@ -46,14 +78,49 @@ func setMariadbPassword() corev1.EnvVar {
 		ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "mariadb-password",
+					Name: secretName,
 				},
-				Key: "password",
+				Key: secretKey,
 			},
 		},
 	}
 }
 
+func generateRandomPassword() string {
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 8
+	buf := make([]rune, length)
+	for i := range buf {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return (string(buf))
+}
+
+func createMariadbPasswordSecret(config *OperatorConfig) error {
+	// Creating a Secret of Type "Opaque" (default)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: config.TargetNamespace,
+		},
+		StringData: map[string]string{
+			secretKey: mariadbPassword,
+		},
+	}
+	//Now convert it to a json file and store locally
+	secretJson, err := json.Marshal(secret)
+	if err != nil {
+		glog.Info("Converting Secret to Json returned error")
+		return err
+	}
+	//Write this to a local file
```

Comment:
What is this for?

Reply:
The goal for this portion of the code was to add the Secret to the openshift-machine-api namespace. I now have a better way of doing this.